### PR TITLE
Add list-group command to display all student groups

### LINKED ISSUE #96: List all the student groups

### PR DESCRIPTION
**Pull request by teekoksiang**
- Add list-group command to display all student groups
- Add test cases for testing `ListGroupCommand`

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListGroupCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    private static final Logger logger = Logger.getLogger(ListGroupCommand.class.getName());
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        assert model != null : "Model is null";
+        requireNonNull(model);
+        List<Person> residentList = model.getFilteredPersonList();
+        logger.log(Level.INFO, "The number of residents is " + residentList.size());
```
Comment (jieffyy) [inline src/main/java/seedu/address/logic/commands/ListGroupCommand.java]: Will this work as intended if the user does:
1. `find n/XX` thereby shortening the list
2. `list-group`

**Review by jieffyy — COMMENTED** [review]
Great work, almost LGTM 👍  Have some clarifications to make regarding the intended usage though.

File: src/main/java/seedu/address/logic/commands/ListGroupCommand.java (hunk 1 above)
Comment (teekoksiang) [reply src/main/java/seedu/address/logic/commands/ListGroupCommand.java]: Good catch, I have updated the logic.

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -91,6 +92,12 @@ public void parseCommand_list() throws Exception {
         assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD + " 3") instanceof ListCommand);
     }
 
+    @Test
+    public void parseCommand_listGroup() throws Exception {
+        assertTrue(parser.parseCommand(ListGroupCommand.COMMAND_WORD) instanceof ListGroupCommand);
+        assertTrue(parser.parseCommand(ListGroupCommand.COMMAND_WORD + " 3") instanceof ListGroupCommand);
```
Comment (jieffyy) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:98]: Hmm this is quite interesting, so if I key `help 3`/`list-group 3`, the command still works - is this a feature or a bug?

We can consider checking for this in another issue.

TLDR: no need to change this part, it's fine!

File: src/test/java/seedu/address/logic/commands/ListGroupCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_listGroup_showsAllStudentGroups() {
+        List<String> groupList = getTypicalStudentGroups();
+        String expectedGroupMessage = formatStudentGroupsWithNumber(groupList);
```
Comment (jieffyy) [inline src/test/java/seedu/address/logic/commands/ListGroupCommandTest.java:34]: Good use of SLAP!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -98,11 +102,13 @@ public boolean hasPerson(Person person) {
     @Override
     public void deletePerson(Person target) {
         addressBook.removePerson(target);
+        personList.remove(target);
     }
 
     @Override
     public void addPerson(Person person) {
         addressBook.addPerson(person);
+        personList.add(person);
```
Comment (jieffyy) [inline src/main/java/seedu/address/model/ModelManager.java]: Hmm this might be a problem because the person is stored in 2 places - an ArrayList and UniquePersonList - complicating the logic. 

E.g, suppose the resident 1 has current student groups basketball. 
1. User adds choir using: `edit 1 s/basketball s/choir`
2. Then searches for all student groups via `list-group`

Will choir still show?

Perhaps you can consider removing this additional field, and then `return addressbook.getPersonList()`? (Check `AddressBook.java` for this method).

Comment (teekoksiang) [reply src/main/java/seedu/address/model/ModelManager.java]: Right, it complicates the logic, I have updated it to `return addressbook.getPersonList()`. Thanks.

**Review by jieffyy — APPROVED** [review]
Nice, LGTM 👍
